osu.Game/Overlays/Music/Playlist.cs

```diff
@@ -30,7 +30,27 @@ public void Filter(FilterCriteria criteria)
             var items = (SearchContainer<RearrangeableListItem<Live<BeatmapSetInfo>>>)ListContainer;
 
             foreach (var item in items.OfType<PlaylistItem>())
-                item.InSelectedCollection = criteria.Collection?.Beatmaps.Any(b => item.Model.ID == b.BeatmapSet?.ID) ?? true;
+            {
+                if (criteria.Collection == null)
+                    item.InSelectedCollection = true;
+                else
+                {
+                    var beatmapHashes = item.Model.Value.Beatmaps.Select(b => b.MD5Hash);
+
+                    bool contained = false;
+
+                    foreach (string hash in beatmapHashes)
+                    {
+                        if (criteria.Collection?.Beatmaps.Contains(hash) == true)
+                        {
+                            contained = true;
+                            break;
+                        }
+                    }
```

Comment:
Is there a reason this is a manual loop rather than something closer to the previous linq expression? Is this done for speed?

Would probably prefer this split out to a separate private method if so.

Reply:
Firstly, the order needed to be reversed to reduce excessive querying of the `Live` (previous method was reading from `Live<>.ID` which was free, but now we are using `MD5Hash`). Secondly, I just found it read better.

Reply:
I think this reads just as well:
```diff
diff --git a/osu.Game/Overlays/Music/Playlist.cs b/osu.Game/Overlays/Music/Playlist.cs
index 1d8163b57b..5354ce0cad 100644
--- a/osu.Game/Overlays/Music/Playlist.cs
+++ b/osu.Game/Overlays/Music/Playlist.cs
@@ -36,19 +36,7 @@ public void Filter(FilterCriteria criteria)
                 else
                 {
                     var beatmapHashes = item.Model.Value.Beatmaps.Select(b => b.MD5Hash);
-
-                    bool contained = false;
-
-                    foreach (string hash in beatmapHashes)
-                    {
-                        if (criteria.Collection?.Beatmaps.Contains(hash) == true)
-                        {
-                            contained = true;
-                            break;
-                        }
-                    }
-
-                    item.InSelectedCollection = contained;
+                    item.InSelectedCollection = beatmapHashes.Any(criteria.Collection.Beatmaps.Contains);
                 }
             }
```
Also, `criteria.Collection` should be guaranteed to be non-null here, right?

Reply:
Yep, will apply something similar to this.